bugsnag-android-core/src/main/java/com/bugsnag/android/Client.java

```diff
@@ -122,7 +124,30 @@ public Client(@NonNull Context androidContext, @NonNull Configuration configurat
         warnIfNotAppContext(androidContext);
         appContext = androidContext.getApplicationContext();
         config = configuration;
-        sessionStore = new SessionStore(config, appContext);
+
+        delegate = new FileStore.Delegate() {
+            @Override
+            public void onErrorIOFailure(Exception exc, File errorFile, String context) {
+                // send an internal error to bugsnag with no cache
+                Thread thread = Thread.currentThread();
+                Error err = new Error.Builder(config, exc, null, thread, true).build();
+                err.setContext(context);
+
+                MetaData metaData = err.getMetaData();
+                metaData.addToTab(INTERNAL_DIAGNOSTICS_TAB, "canRead", errorFile.canRead());
+                metaData.addToTab(INTERNAL_DIAGNOSTICS_TAB, "canWrite", errorFile.canWrite());
+                metaData.addToTab(INTERNAL_DIAGNOSTICS_TAB, "exists", errorFile.exists());
+
+                @SuppressLint("UsableSpace") // requires API 26
+                long usableSpace = appContext.getCacheDir().getUsableSpace();
+                metaData.addToTab(INTERNAL_DIAGNOSTICS_TAB, "usableSpace", usableSpace);
+                metaData.addToTab(INTERNAL_DIAGNOSTICS_TAB, "filename", errorFile.getName());
+                metaData.addToTab(INTERNAL_DIAGNOSTICS_TAB, "fileLength", errorFile.length());
+                Client.this.reportInternalBugsnagError(err);
+            }
+        };
+
+        sessionStore = new SessionStore(config, appContext, delegate);
```

Comment:
Should this be done at all, or even be done differently for the session store? It has the potential to be much more verbose since every app launch sends a session rather than just error prone ones, and the context values should be different from "Crash Report ..." which is specified in `FileStore`.

Reply:
Avoiding this for now due to the volume of the sessions vs errors sounds like a good call - I'll update the PR to pass in null as the delegate, which will prevent internal error reporting.